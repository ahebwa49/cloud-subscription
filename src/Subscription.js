import React, { useState } from "react";
import "./Subscription.scss";

const Subscription = () => {
  const [duration, setDuration] = useState("3");
  const [storage, setStorage] = useState("5");
  const [upfrontPayment, setUpfrontPayment] = useState("No");
  const [step, setStep] = useState(1);

  const onDurationValueChange = (e) => {
    console.log(e.target.value);
    setDuration(e.target.value);
  };

  const onStorageValueChange = (e) => {
    console.log(e.target.value);
    setStorage(e.target.value);
  };

  const onUpfrontPaymentValueChange = (e) => {
    console.log(e.target.value);
    setUpfrontPayment(e.target.value);
  };

  return (
    <div className="Subscription">
      <h3>Subscription Parameters</h3>
      <form className="SubscriptionForm">
        <div className="SubscriptionFormGroup">
          <label htmlFor="duration" className="SubscriptionFormGroup__label">
            Duration
          </label>
          <input
            type="radio"
            name="duration"
            value="3"
            onChange={onDurationValueChange}
            checked={duration === "3"}
          />{" "}
          3 months
          <input
            type="radio"
            name="duration"
            value="6"
            onChange={onDurationValueChange}
            checked={duration === "6"}
          />{" "}
          6 months
          <input
            type="radio"
            name="duration"
            value="12"
            onChange={onDurationValueChange}
            checked={duration === "12"}
          />{" "}
          12 months
        </div>
        <div className="SubscriptionFormGroup">
          <label htmlFor="storage" className="SubscriptionFormGroup__label">
            Amount of gigabytes in a cloud
          </label>
          <input
            type="radio"
            name="storage"
            value="5"
            onChange={onStorageValueChange}
            checked={storage === "5"}
          />{" "}
          5 GB
          <input
            type="radio"
            name="storage"
            value="10"
            onChange={onStorageValueChange}
            checked={storage === "10"}
          />{" "}
          10 GB
          <input
            type="radio"
            name="storage"
            value="50"
            onChange={onStorageValueChange}
            checked={storage === "50"}
          />{" "}
          50 GB
        </div>
        <div className="SubscriptionFormGroup">
          <label
            htmlFor="upfrontPayment"
            className="SubscriptionFormGroup__label"
          >
            Upfront payment:
          </label>
          <input
            type="radio"
            name="upfrontPayment"
            value="No"
            onChange={onUpfrontPaymentValueChange}
            checked={upfrontPayment === "No"}
          />{" "}
          No
          <input
            type="radio"
            name="upfrontPayment"
            value="Yes"
            onChange={onUpfrontPaymentValueChange}
            checked={upfrontPayment === "Yes"}
          />{" "}
          Yes
        </div>
      </form>
    </div>
  );
};
export default Subscription;
