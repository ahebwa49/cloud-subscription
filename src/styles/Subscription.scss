.Subscription {
  display: flex;
  flex-direction: column;
  margin-top: 150px;
  padding: 1rem;
  padding-top: 0;
  width: 450px;
  height: 330px;
  border-radius: 4px;
  border: 1px solid #DADADA;
}
.SubscriptionFormGroup{
    margin-bottom: 1rem;
}
