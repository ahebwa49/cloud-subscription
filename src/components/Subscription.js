import React, { useState } from "react";
import "../styles/Subscription.scss";

const Subscription = ({
  onDurationValueChange,
  onStorageValueChange,
  onUpfrontPaymentValueChange,
  plans,
  duration,
  storage,
  upfrontPayment,
}) => {
  const storagePlans = [5, 10, 50];
  return (
    <div className="Subscription">
      <h4>Subscription Parameters</h4>
      <form className="SubscriptionForm">
        <div className="SubscriptionFormGroup">
          <label htmlFor="duration" className="SubscriptionFormGroup__label">
            Duration:
          </label>
          <br />
          {plans.map((plan, index) => (
            <div key={index}>
              <input
                key={index}
                required
                type="radio"
                name="duration"
                value={plan.duration_months}
                onChange={(e) => onDurationValueChange(e, plan)}
                checked={duration === plan.duration_months}
              />
              <span>
                {plan.duration_months} months, price usd per gb:{" "}
                {plan.price_usd_per_gb}
              </span>
              <br />
            </div>
          ))}
        </div>
        <div className="SubscriptionFormGroup">
          <label htmlFor="storage" className="SubscriptionFormGroup__label">
            Amount of gigabytes in a cloud:
          </label>
          <br />
          {storagePlans.map((plan, index) => (
            <div key={index}>
              <input
                key={index}
                type="radio"
                name="storage"
                value={plan}
                required
                onChange={(e) => onStorageValueChange(e, plan)}
                checked={storage === plan}
              />
              <span>{plan} GB</span>
              <br />
            </div>
          ))}
        </div>
        <div className="SubscriptionFormGroup">
          <label
            htmlFor="upfrontPayment"
            className="SubscriptionFormGroup__label"
          >
            Upfront payment:
          </label>
          <input
            type="radio"
            name="upfrontPayment"
            required
            value="No"
            onChange={(e) => onUpfrontPaymentValueChange(e)}
            checked={upfrontPayment === "No"}
          />{" "}
          No
          <input
            type="radio"
            name="upfrontPayment"
            value="Yes"
            required
            onChange={(e) => onUpfrontPaymentValueChange(e)}
            checked={upfrontPayment === "Yes"}
          />{" "}
          Yes
        </div>
      </form>
    </div>
  );
};
export default Subscription;
