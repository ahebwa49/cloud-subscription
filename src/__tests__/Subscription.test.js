import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Subscription from "../components/Subscription";
import subscriptionData from "./data/subscriptionData";

describe("create subscription snapshot", () => {
  test("create snapshot", () => {
    let tree = renderer
      .create(
        <Subscription
          plans={[
            { duration_months: 3, price_usd_per_gb: 3 },
            { duration_months: 6, price_usd_per_gb: 2.5 },
            { duration_months: 12, price_usd_per_gb: 2 },
          ]}
          onDurationValueChange={() => {}}
          onStorageValueChange={() => {}}
          onUpfrontPaymentValueChange={() => {}}
          storage={5}
          duration={6}
          upfrontPayment="Yes"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

let tree;

for (let data of subscriptionData) {
  let { plans, storage, duration } = data;
  describe("Consistency in subscription component", () => {
    beforeEach(() => {
      tree = render(
        <Subscription
          plans={plans}
          onDurationValueChange={() => {}}
          onStorageValueChange={() => {}}
          onUpfrontPaymentValueChange={() => {}}
          storage={storage}
          duration={duration}
          upfrontPayment="Yes"
        />
      );
    });
    test("Ensure that all duration radio buttons are displayed in the UI", () => {
      let nodeList = document.querySelectorAll(
        ".SubscriptionFormGroup input[name=duration]"
      );
      expect(nodeList.length).toBe(plans.length);
    });
    test("Ensure that all storage radio buttons are displayed in the UI", () => {
      let nodeList = document.querySelectorAll(
        ".SubscriptionFormGroup input[name=storage]"
      );
      expect(nodeList.length).toBe(3);
    });
    test("Ensure that all the upfront radio buttons are displayed in the UI", () => {
      let nodeList = document.querySelectorAll(
        ".SubscriptionFormGroup input[name=upfrontPayment]"
      );
      expect(nodeList.length).toBe(2);
    });
  });
}
