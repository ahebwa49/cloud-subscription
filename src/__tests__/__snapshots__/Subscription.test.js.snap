// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create subscription snapshot create snapshot 1`] = `
<div
  className="Subscription"
>
  <h4>
    Subscription Parameters
  </h4>
  <form
    className="SubscriptionForm"
  >
    <div
      className="SubscriptionFormGroup"
    >
      <label
        className="SubscriptionFormGroup__label"
        htmlFor="duration"
      >
        Duration:
      </label>
      <br />
      <div>
        <input
          checked={false}
          name="duration"
          onChange={[Function]}
          required={true}
          type="radio"
          value={3}
        />
        <span>
          3
           months, price usd per gb:
           
          3
        </span>
        <br />
      </div>
      <div>
        <input
          checked={true}
          name="duration"
          onChange={[Function]}
          required={true}
          type="radio"
          value={6}
        />
        <span>
          6
           months, price usd per gb:
           
          2.5
        </span>
        <br />
      </div>
      <div>
        <input
          checked={false}
          name="duration"
          onChange={[Function]}
          required={true}
          type="radio"
          value={12}
        />
        <span>
          12
           months, price usd per gb:
           
          2
        </span>
        <br />
      </div>
    </div>
    <div
      className="SubscriptionFormGroup"
    >
      <label
        className="SubscriptionFormGroup__label"
        htmlFor="storage"
      >
        Amount of gigabytes in a cloud:
      </label>
      <br />
      <div>
        <input
          checked={true}
          name="storage"
          onChange={[Function]}
          required={true}
          type="radio"
          value={5}
        />
        <span>
          5
           GB
        </span>
        <br />
      </div>
      <div>
        <input
          checked={false}
          name="storage"
          onChange={[Function]}
          required={true}
          type="radio"
          value={10}
        />
        <span>
          10
           GB
        </span>
        <br />
      </div>
      <div>
        <input
          checked={false}
          name="storage"
          onChange={[Function]}
          required={true}
          type="radio"
          value={50}
        />
        <span>
          50
           GB
        </span>
        <br />
      </div>
    </div>
    <div
      className="SubscriptionFormGroup"
    >
      <label
        className="SubscriptionFormGroup__label"
        htmlFor="upfrontPayment"
      >
        Upfront payment:
      </label>
      <input
        checked={false}
        name="upfrontPayment"
        onChange={[Function]}
        required={true}
        type="radio"
        value="No"
      />
       
      No
      <input
        checked={true}
        name="upfrontPayment"
        onChange={[Function]}
        required={true}
        type="radio"
        value="Yes"
      />
       
      Yes
    </div>
  </form>
</div>
`;
